### ISSUE TITLE
Gradient Penalty Loss: Not working with tf2

### ISSUE THREAD

**Issue by pabloi09** [eriklindernoren/Keras-GAN#215]
Good morning,
The aproach of the gradient_penalty_loss in the wgan_gp is not working with tf2. K.gradients returns [None]. You guys can find the issue here:
https://colab.research.google.com/drive/11dcMKoiCigTnEn7QvmjqLNrJdmFztByT#scrollTo=-cI983tc_XC7
I have been reading some posts and it seems like there is no solution yet. Is there anyone that has already implemented this loss successfully with tf2?
Thank you very much

**Comment by cckuailong** [comment]
> Good morning,
> The approach of the gradient_penalty_loss in the wgan_gp is not working with tf2. K.gradients returns [None]. You guys can find the issue here:
> https://colab.research.google.com/drive/11dcMKoiCigTnEn7QvmjqLNrJdmFztByT
> I have been reading some posts and it seems like there is no solution yet. Is there anyone that has already implemented this loss successfully with tf2?
> Thank you very much

Maybe we can use tf.keras.backend.function to solve the problem. I write a sample version of WGAN-GP [Here](https://github.com/cckuailong/KerGaNs/tree/master/wgan_gp). It's tested on the newest tf.keras.

**Comment by pabloi09** [comment]
> Maybe we can use tf.keras.backend.function to solve the problem. I write a sample version of WGAN-GP [Here](https://github.com/cckuailong/KerGaNs/tree/master/wgan_gp). It's tested on the newest tf.keras.

Hello! 

Your solution seems pretty smart. I didn't get to that and I ended doing the models extending tf.keras.Model and using GradientTape. This seems to be the standard now with tf2. The example I followed  can be found <a href="https://github.com/timsainb/tensorflow2-generative-models">Here</a>

**Comment by cckuailong** [comment]
> > Maybe we can use tf.keras.backend.function to solve the problem. I write a sample version of WGAN-GP [Here](https://github.com/cckuailong/KerGaNs/tree/master/wgan_gp). It's tested on the newest tf.keras.
> 
> Hello!
> 
> Your solution seems pretty smart. I didn't get to that and I ended doing the models extending tf.keras.Model and using GradientTape. This seems to be the standard now with tf2. The example I followed can be found [Here](https://github.com/timsainb/tensorflow2-generative-models)

Nice job! I think Keras will add the GradientTape soonly.

**Comment by Shuhain1994** [comment]
can anyone write it with GradientTape?
